Makefile

```diff
@@ -1,10 +1,42 @@
-setup:
-	curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
-	pyenv install '2.7.14'
-	pyenv local 2.7.14
-	pyenv virtualenv 2.7.14 samtranslator27
-	pyenv activate samtranslator27
-	pyenv local samtranslator27
+NAME=samtranslator27
+PYTHON_VERSION=2.7.14
+PYENV := $(shell command -v pyenv 2> /dev/null)
+UNAME := $(shell uname -s)
+
+ifeq ($(UNAME),Darwin)
+	# Mac OS X https://github.com/pyenv/pyenv/wiki/Common-build-problems#build-failed-error-the-python-zlib-extension-was-not-compiled-missing-the-zlib
+	CFLAGS="-O2 -I$$(xcrun --show-sdk-path)/usr/include"
+else
+	# See: https://github.com/pyenv/pyenv/wiki/Common-build-problems#make-your-pythons-a-little-faster
+	CFLAGS="-O2"
+endif
+
+# Make sure that pyenv is configured properly and that we can use it in our setup target.
+validation:
+ifndef PYENV
+    $(error "make sure pyenv is accessible in your path, (usually by adding to PATH variable in bash_profile, zshrc, or other locations based on your platform) See: https://github.com/pyenv/pyenv#installation for the installation insructions.")
+endif
+ifndef PYENV_SHELL
+	$(error "Add 'pyenv init' to your shell to enable shims and autocompletion, (usually by adding to your bash_profile, zshrc, or other locations based on your platform)")
+endif
+ifndef PYENV_VIRTUALENV_INIT
+	$(error "Add 'pyenv virtualenv-init' to your shell to enable shims and autocompletion, (usually by adding to your bash_profile, zshrc, or other locations based on your platform)")
+endif
+
+# Setup the pyenv environment
+setup: validation
+	CFLAGS=$(CFLAGS) pyenv install $(PYTHON_VERSION)
+	pyenv local $(PYTHON_VERSION) 
+	pyenv virtualenv $(PYTHON_VERSION) $(NAME)
+	make activate
+	
+# Activate the environment
+activate:
+	$(shell eval "$$(pyenv init -)" && eval "$$(pyenv virtualenv-init -)" && pyenv activate $(NAME) && pyenv local $(NAME))
+	
+# Remove all pyenv versions to start fresh	
+clean:
```

Comment:
Can we add pyenv to this command? I can imagine a clean command that cleans all .pyc files or tox. It will make this command a little clearer on what will happen.

Reply:
Not sure how to clean tox things but added the pyc and python cache folders